controller/src/main/java/org/jboss/as/controller/persistence/yaml/YamlConfigurationExtension.java

```diff
@@ -524,6 +549,25 @@ public String getCommandLineInstructions() {
         return MGMT_OP_LOGGER.argYaml();
     }
 
+    @SuppressWarnings("unchecked")
+    private void processUnmanagedDeployments(ImmutableManagementResourceRegistration rootRegistration, Map.Entry<String, Object> deployment, Map<PathAddress, ParsedBootOp> xmlOperations, List<ParsedBootOp> postExtensionOps) {
+        String name = deployment.getKey();
+        OperationEntry operationEntry = rootRegistration.getOperationEntry(PathAddress.pathAddress("deployment", name), ADD);
+        if (deployment.getValue() != null && deployment.getValue() instanceof Map) {
```

Comment:
Nit:

This can be

if (deployment.getValue() instanceof Map) {

or, since this check was already performed in load(), simply

assert deployment.getValue() instanceof Map;